vassal-doc/src/main/readme-referencemanual/ReferenceManual/ExpressionSumCount.adoc

```diff
@@ -41,134 +53,863 @@ See <<PropertyMatchExpression.adoc#top>> for more information.
+|===
+|*SumRangePx* |Returns the total of the named property in pieces within range (in pixels) of the current piece +
+
+|SumRangePx(min, max, prop)
+
+#[ Trait Only ]#| Sum a property across all pieces within the specified range in Pixels +
+
+[example]
+SumRangePx(1, 100, "Strength") +
+
+Sum the value of the property _Strength_ in all pieces in within 100 pixels of the current piece, but excluding the current piece and any piece stacked in the same location as the current piece
+
+|SumRangePx(min, max, prop, expr)
+
+#[ Trait Only ]#| Sum a property across all pieces within the specified range in grid units AND that match the supplied <<PropertyMatchExpression.adoc#top,Property Match Expression>>. +
+
+[example]
+SumRangePx(0, 500, "Strength", "{Alien==true}") +
+
+Sum the value of the property _Strength_ in pieces within 500 pixels that have a value of *true** for the _Alien_ property.
+|===
+
+
+[#CountRange]
+[width=100%,cols="33%a,67%a"]
+|===
+|*CountRange* |Count the number of pieces within range (in Grid Units) of the current piece +
+
+|CountRange(min, max)
+
+#[ Trait Only ]#| Count all pieces within the specified range in Grid Units +
+
+[example]
+CountRange(1, 5) +
+
+Count all pieces within 5 hexes of the current piece,
+but not in the same hex as the current piece.
+
+|CountRange(min, max, prop)
+
+#[ Trait Only ]#| Count all pieces within the specified range in Grid Units AND that have a non-blank value for the named property.
+
+[example]
+CountRange(1, 5, "Russian") +
+
+Count all pieces within 5 hexes of the current piece,
+but not in the same hex as the current piece and that have a non-blank value for the _Russian_ property.
+
+|CountRange(min, max, expr)
+
+#[ Trait Only ]#| Count all pieces within the specified range in Grid Units that match the supplied <<PropertyMatchExpression.adoc#top,Property Match Expression>>. +
+
+[example]
+CountRange(1, 5, "{Army==\"Russian\"}") +
+
+Count all pieces within 5 hexes of the current piece,
+but not in the same hex as the current piece and that have a value of "Russian" for the _Army_ property.
+
+
+|CountRange(min, max, prop, expr)
+
+#[ Trait Only ]#| Count all pieces within the specified range in grid units that have a non-blank value for the named property AND that match the supplied <<PropertyMatchExpression.adoc#top,Property Match Expression>>. +
+
+[example]
+CountRange(0, 2, "Isolated", "{Army==\"German\"}") +
+
+Cout the pieces within 2 hexes of the current piece that have a non-blank value for the property _Isolated_ AND that have a value of "German" for the _Army_ property.
+|===
+
+
+[#CountRangePx]
+[width=100%,cols="33%a,67%a"]
+|===
+|*CountRangePx* |Count the number of pieces within range (in pixels) of the current piece +
+
+|CountRangePx(min, max)
+
+#[ Trait Only ]#| Count all pieces within the specified range in pixels +
+
+[example]
+CountRangePx(1, 5) +
+
+Count all pieces within 5 hexes of the current piece,
+but not in the same hex as the current piece.
```

Comment:
but not in the same hex => but excluding the current piece and any piece stacked in the same location as the current piece